processing/src/main/java/io/druid/query/dimension/DimensionSpec.java

```diff
@@ -22,23 +22,30 @@
 import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import io.druid.query.extraction.ExtractionFn;
+import io.druid.segment.DimensionSelector;
 
 /**
  */
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = LegacyDimensionSpec.class)
 @JsonSubTypes(value = {
     @JsonSubTypes.Type(name = "default", value = DefaultDimensionSpec.class),
-    @JsonSubTypes.Type(name = "extraction", value = ExtractionDimensionSpec.class)
+    @JsonSubTypes.Type(name = "extraction", value = ExtractionDimensionSpec.class),
+    @JsonSubTypes.Type(name = "regexFiltered", value = RegexFilteredDimensionSpec.class),
```

Comment:
These are only used for multi-valued dimensions, as the docs say?  I understand that there are a number of things to keep straight, but what are you concretely suggesting we do differently?

I think everyone would be open to a proposal for a new query path, if you have concrete ideas on how to improve that, maybe this type of filtering wouldn't be required in that world?  Or are you asking for javadoc on the classes to clarify when to use them?